subprojects/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java

```diff
@@ -218,6 +223,15 @@ public long getAgeMillis(BuildCommencedTimeProvider timeProvider, CachedExternal
         return timeProvider.getCurrentTime() - cached.getCachedAt();
     }
 
+    public boolean isResourceStillValid(BuildCommencedTimeProvider timeProvider, CachedExternalResource cached) {
```

Comment:
This should be part of the call to `externalResourceCachePolicy.mustRefreshExternalResource`, I think.